nav2_collision_monitor/include/nav2_collision_monitor/polygon.hpp

```diff
@@ -145,6 +145,19 @@ class Polygon
    */
   virtual int getPointsInside(const std::vector<Point> & points) const;
 
+  /**
+   * @brief Add a new decision point to count the consecutive number of
+   * time enough point are in the polygon. Updates the current decision value
+   * based on previous measurements.
+   */
+  void newDecisionPoint(bool is_in);
```

Comment:
On naming: this is a little ambiguous as to the what decision it is. Can you come up with some alternatives?